dhis-2/dhis-web-api-test/src/test/java/org/hisp/dhis/webapi/controller/StaticContentControllerTest.java

```diff
@@ -206,7 +206,7 @@ void verifyErrorWhenStoringInvalidMimeType()
         final String error = buildResponse( "Unsupported Media Type", 415, "WARNING", null );
         mvc.perform( multipart( URL + LOGO_BANNER )
             .file( new MockMultipartFile( "file", "testlogo.png", IMAGE_JPEG.toString(), "image".getBytes() ) )
-            .session( session ) ).andExpect( content().json( error ) )
+            .session( session ) ).andExpect( content().string( error ) )
```

Comment:
String comparison of 2 JSON sounds like it could lead to a brittle test just because of whitespace issues. I think this here can be replaced with a json tree assertion? Like we do in the convenience controller tests. `buildResponse` builds the expected JSON using Gson. So we can replace that with json-tree.

Reply:
It makes sense.
I am focusing just on the dependencies now, I am putting back the jsonassert dependency and add a comment explaining what is the issue